### PR TITLE
Read automation information from journal.

### PR DESCRIPTION
**Pull request by ikeough**
### Purpose

This PR augments the ability to run Dynamo from a journal file. Previously, you could specify a Dynamo workspace to open from a journal file. But, because `IsTestMode` was set to `false`, a workflow once opened, even in automatic mode, would not run as the Dynamo Revit application would never go into idling mode.  Now `DynamoRevit.Execute` does the following:
- Checks the journal file's `dynShowUI` key and sets a private static field to true or false.
- Checks the journal's `dynAutomation` key and sets `IsTestMode` to correspond. `IsTestMode` is set to `false` if this key is not present or the journal file data is null.
- Checks the journal's `dynPath` key. If it exists and points to a proper path, it then opens (and runs) the workspace. 

### Declarations

Check these if you believe they are true

- [x] The code base is in a better state after this PR
  - Added null checks and block formatting.
- [x] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
  - No new user facing strings introduced.

### Reviewers

@Randy-Ma This PR is for work not in the current sprint, but it is in support of @andydandy74's RTCEurope class "Automating Automation". I'd appreciate your quick review.

### FYIs

@andydandy74

### Cherry Picks:
- [ ] Revit2015

### CODE REVIEW

File: src/DynamoRevit/DynamoRevit.cs [hunk 1]
```diff
@@ -414,4 +460,17 @@ public override IEnumerable<string> GetDynamoInstallLocations()
                 (s) => regKey.OpenSubKey(s).GetValue("InstallLocation") as string);
         }
     }
+
+    internal class AutomationThread : ISchedulerThread
```
Comment (Randy-Ma) [inline src/DynamoRevit/DynamoRevit.cs]: It seems this scheduler thread is temporarily not used.

File: src/DynamoRevit/DynamoRevit.cs [hunk 2]
```diff
@@ -281,11 +291,47 @@ private void HandleDebug(ExternalCommandData commandData)
...
+        private static bool ShowUI(ExternalCommandData commandData)
+        {
+            if (commandData.JournalData == null)
+            {
+                return true;
+            }
+
+            if (commandData.JournalData.ContainsKey("dynShowUI"))
+            {
+                return bool.Parse(commandData.JournalData["dynShowUI"]);
```
Comment (Randy-Ma) [inline src/DynamoRevit/DynamoRevit.cs]: TryParse is better so that there will be no exceptions thrown.

File: src/DynamoRevit/DynamoRevit.cs [hunk 3]
```diff
@@ -281,11 +291,47 @@ private void HandleDebug(ExternalCommandData commandData)
...
+        private static bool IsAutomationMode(ExternalCommandData commandData)
+        {
+            if (commandData.JournalData == null)
+            {
+                return false;
+            }
+
+            if (commandData.JournalData.ContainsKey("dynAutomation"))
+            {
+                return bool.Parse(commandData.JournalData["dynAutomation"]);
```
Comment (Randy-Ma) [inline src/DynamoRevit/DynamoRevit.cs]: Similarly TryParse is better

**Comment by Randy-Ma** [comment]
This is awesome! Except for some minor suggestions, LGTM

File: src/DynamoRevit/DynamoRevit.cs (hunk 2 above)
Comment (ikeough) [inline src/DynamoRevit/DynamoRevit.cs]: Very good point @Randy-Ma. Will do.

File: src/DynamoRevit/DynamoRevit.cs (hunk 1 above)
Comment (ikeough) [inline src/DynamoRevit/DynamoRevit.cs]: I mistakenly left this class in. It's not used. I'll remove it.

**Pull request merged**